extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -352,4 +357,12 @@ export class AmpAd3PImpl extends AMP.BaseElement {
         overflown, requestedHeight, requestedWidth);
     }
   }
+
+  /** Function to collapse parent <amp-ad> element*/
+  collapseParent() {
+    this.resources_.attemptChangeSize(this.parent_, 0, 0, () => {
```

Comment:
You also need to fix
https://github.com/ampproject/amphtml/blob/master/extensions/amp-ad/0.1/amp-ad-api-handler.js#L147

Maybe you can actually fix both by overriding `attemptChangeHeight` in this class to operate on the parent.